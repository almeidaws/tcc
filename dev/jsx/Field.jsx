import React from 'react';

/**
 * This field was created to be used on modals used to register
 * and login the user. It contains a set os props and also allow
 * set a error message. It doesn't have state.
 *
 * Property type: the same from HTML elements. Examples: 'email', 'password'.
 * Property placeholder: the same from HTML elements. Example: 'Your email...'.
 * Property value: the current value of this field. It's a controlled component.
 * Property onChange: callback called when the field's value changes. The generated
 * event is sent as argument.
 * Property required: 'true' if the field must contains some value or 'false' otherwise.
 * Property icon: icon's name rendered on the field. This name are that used on bootstrap.
 * Example: 'evelope', 'lock', etc.
 * Property errorMessage: If this value is setted, an error message is presented below
 * the field. If there's no error this the value must be some JavaScript's falsy value
 * like 'null' of 'undefined'.
 */
const Field = props => (
    <div className="form-group">
        <input 
            type={props.type} 
            placeholder={props.placeholder} 
            className="form-control" 
            value={props.value} 
            onChange={props.onChange}
            required={props.required}/>
        <span className="form_icon">
            <i className={"fa_icon form-" + props.icon} aria-hidden="true" />
        </span>
        <div className="invalid-feedback text-warning">{props.errorMessage}</div>
    </div>
);

export default Field;
