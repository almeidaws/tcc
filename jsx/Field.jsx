const Field = props => (
    <div className="form-group">
        <input 
            type={props.type} 
            placeholder={props.email} 
            className="form-control" 
            value={props.value} 
            onChange={props.onChange}
            required={props.required}/>
        <span className="form_icon">
            <i className={"fa_icon form-" + props.icon} aria-hidden="true" />
        </span>
        <div className="invalid-feedback">{props.errorMessage}</div>
    </div>
);
