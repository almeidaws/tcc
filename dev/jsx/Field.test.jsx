import React from 'react';
import Field from './Field.jsx';
import renderer from 'react-test-renderer';

it('<Field /> should render email', () => {
    const component = renderer.create(<Field type='email'
                                             placeholder='example@gmail.com'
                                             className='form-control'
                                             value='paulo@gmail.com' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('<Field /> should present an error', () => {
    const component = renderer.create(<Field type='email'
                                             placeholder='example@gmail.com'
                                             className='form-control'
                                             value='paulo@gmail.com'
                                             errorMessage='Invalid email' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

