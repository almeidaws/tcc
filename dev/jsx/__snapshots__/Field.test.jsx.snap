// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> should present an error 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    placeholder="example@gmail.com"
    type="email"
    value="paulo@gmail.com"
  />
  <span
    className="form_icon"
  >
    <i
      aria-hidden="true"
      className="fa_icon form-undefined"
    />
  </span>
  <div
    className="invalid-feedback text-warning"
  >
    Invalid email
  </div>
</div>
`;

exports[`<Field /> should render email 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    placeholder="example@gmail.com"
    type="email"
    value="paulo@gmail.com"
  />
  <span
    className="form_icon"
  >
    <i
      aria-hidden="true"
      className="fa_icon form-undefined"
    />
  </span>
  <div
    className="invalid-feedback text-warning"
  />
</div>
`;
